lib/entities/semantic-file.js

```diff
@@ -0,0 +1,161 @@
+import { sparqlEscape } from 'mu';
+import { SERVICE_URI } from '../../env';
+import { filenameToUri, uriToFilename } from '../util/file';
+import { v5 as uuidv5 } from 'uuid';
+import path from 'path';
+import fs from 'fs-extra';
+import Base64 from 'crypto-js/enc-base64';
+import sha256 from 'crypto-js/sha256';
+import moment from 'moment';
+
+const NAMESPACE = '77219b1e-e23d-4828-847f-55e2d7fac687'; // NOTE: used for consistent uuid generation based on the URI
+
+/**
+ * Class representation of a semantic-file
+ */
+export class SemanticFile {
+
+  static exists(filename) {
+    if (filename instanceof SemanticFile) {
+      filename = filename.filename;
+    }
```

Comment:
Could we not directly passing the semanticFile's filename to the function ? In the end it's only used in the case where you check if a semanticFile exists no ?

Reply:
It was a little experiment, to make it possible to reuse this function as a helper without the need to create an entire semantic-file. It's is maybe a little redundant right now, but could be useful in the future without the need to define an explicit helper function.